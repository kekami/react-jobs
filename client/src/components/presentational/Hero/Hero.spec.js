import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Hero } from './Hero';

describe('Hero Component tests.', () => {
  let wrapper;

  it('Renders without crashing.', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Hero />
      </MemoryRouter>, div);
  });

  describe('Detailed snapshot.', () => {
    beforeEach(() => {
      wrapper = shallow(<Hero />);
    });

    it('Matches standard output.', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Contains one Centered styled tag for the body.', () => {
      expect(wrapper.find('Centered').length).toBe(1);
    });

    it('Contains one Footer styled tag for the body.', () => {
      expect(wrapper.find('Footer').length).toBe(1);
    });
  });
});
