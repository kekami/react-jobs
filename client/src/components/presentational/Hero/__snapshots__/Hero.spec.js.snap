// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component tests. Detailed snapshot. Matches standard output. 1`] = `
<styled.div>
  <Centered>
    <SVG />
    <SVG />
    <styled.h2>
      Travel. Work.
    </styled.h2>
    <styled.h1>
      <em>
        See the world
      </em>
    </styled.h1>
    <styled.div>
      <Link
        replace={false}
        to="/jobs"
      >
        <styled.button>
          <styled.i
            aria-hidden="true"
            className="fa fa-search fa-2x"
          />
          EXPLORE ALL JOBS
        </styled.button>
      </Link>
      <styled.button>
        <styled.i
          aria-hidden="true"
          className="fa fa-users fa-2x"
        />
        HIRING? START HERE
      </styled.button>
    </styled.div>
    <styled.p>
      Use Jobbatical to find a tech, business, 
      <br />
      or creative job abroad.
    </styled.p>
  </Centered>
  <Footer>
    <styled.p>
      Currently heaps of teams in loads of 
      <br />
      countries could use your help! 
    </styled.p>
  </Footer>
</styled.div>
`;
